starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -74,8 +76,10 @@ public Optional<Inventory> retrieve(String id) {
    * @return Updated Inventory.
    */
   public Optional<Inventory> update(String id, Inventory inventory) {
-    // TODO
-    return Optional.empty();
+    Query query = new Query();
+    query.addCriteria(Criteria.where("id").is(id));
+    Inventory newInventory = this.mongoTemplate.findAndReplace(query, inventory, options().returnNew(), "inventory");
```

Comment:
Should probably use findAndModify, you want to preserve the original id, and just update the existing inventory not completely replace it with a new inventory.